KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/RecordingModule.java

```diff
@@ -305,10 +307,9 @@ public void onPublish(IMediaStream stream, String streamName, boolean isRecord,
             KalturaEntryServerNodeType serverIndex;
 
             // This code section should run for source steam that has recording
-            synchronized(properties) {
-                 serverIndex = KalturaEntryServerNodeType.get(properties.getPropertyStr(Constants.CLIENT_PROPERTY_SERVER_INDEX, Constants.INVALID_SERVER_INDEX));
-            }
-
+            String serverId = (String) entryMap.get(Constants.CLIENT_PROPERTY_SERVER_INDEX);
```

Comment:
changes done in Utils.java, AuthenticationModule.java and RecordingModule.java. Need to review carefully.